aws/aws-parallelcluster
# Fix custom cookbook S3 url

Fix custom cookbook s3 url not work 
The presign url not work previously because of:
lacking instance policy to presign s3 url 
aws presign ... --region should be bucket region instead of instance region

Add policy to presign url and get bucket region
Retrieve bucket region instead of using instance region
Remove not used parts when presign s3 url in the code to fix error: Warning: Illegal date format for -z, --time-cond (and not a file name).

Manually tests:
Before adding policy:
```
cat: /etc/chef/aws-parallelcluster-cookbook.tgz.date: No such file or directory
Warning: Illegal date format for -z, --time-cond (and not a file name). 
< HTTP/1.1 403 Forbidden
```
After adding IAM policy:
```
< HTTP/1.1 200 OK
```
Add a cookbook block in the manually test custom cookbook:
execute 'test custom cookbook' do
  command "echo 'This is use to test custom cookbook'"
end
I am able to see the execution result in chef-client.log:
```
execute] This is use to test custom cookbook
[2021-08-09T21:21:01+00:00] INFO: execute[test custom cookbook] ran successfully
    - execute echo 'This is use to test custom cookbook'
    - ```
